notebook/static/base/js/utils.js

```diff
@@ -192,165 +193,231 @@ define([
-                    g = Math.floor(idx / 6) % 6;
-                    r = Math.floor(idx / 36) % 6;
-                    // convert to rgb
-                    r = (r * 51);
-                    g = (g * 51);
-                    b = (b * 51);
-                } else {
-                    // grayscale
-                    idx = idx - 231;
-                    // it's 1-24 and should *not* include black or white,
-                    // so a 26 point scale
-                    r = g = b = Math.floor(idx * 256 / 26);
-                }
-            } else if (index_or_rgb == "2") {
-                // Simple 24 bit RGB
-                if (numbers.length > 3) {
-                    console.log("Not enough fields for RGB", numbers);
-                    return;
-                }
-                r = numbers.shift();
-                g = numbers.shift();
-                b = numbers.shift();
-            } else {
-                console.log("unrecognized control", numbers);
-                return;
+        }
+        return numbers;
+    }
+
+    function _getExtendedColors(numbers) {
+        var r, g, b;
+        var n = numbers.shift();
+        if (n === 2 && numbers.length >= 3) {
+            // 24-bit RGB
+            r = numbers.shift();
+            g = numbers.shift();
+            b = numbers.shift();
+            if ([r, g, b].some(function (c) { return c < 0 || 255 < c; })) {
+                throw new RangeError();
             }
-            if (r !== undefined) {
-                // apply the rgb color
-                var line;
-                if (n == "38") {
-                    line = "color: ";
-                } else {
-                    line = "background-color: ";
-                }
-                line = line + "rgb(" + r + "," + g + "," + b + ");";
-                if ( !attrs.style ) {
-                    attrs.style = line;
-                } else {
-                    attrs.style += " " + line;
-                }
+        } else if (n === 5 && numbers.length >= 1) {
+            // 256 colors
+            var idx = numbers.shift();
+            if (idx < 0) {
+                throw new RangeError();
+            } else if (idx < 16) {
+                // 16 default terminal colors
+                return idx;
+            } else if (idx < 232) {
+                // 6x6x6 color cube, see http://stackoverflow.com/a/27165165/500098
+                r = Math.floor((idx - 16) / 36);
+                r = r > 0 ? 55 + r * 40 : 0;
+                g = Math.floor(((idx - 16) % 36) / 6);
+                g = g > 0 ? 55 + g * 40 : 0;
+                b = (idx - 16) % 6;
+                b = b > 0 ? 55 + b * 40 : 0;
+            } else if (idx < 256) {
+                // grayscale, see http://stackoverflow.com/a/27165165/500098
+                r = g = b = (idx - 232) * 10 + 8;
+            } else {
+                throw new RangeError();
             }
+        } else {
+            throw new RangeError();
```

Comment:
It's a good idea to be informative on these RangeErrors, indicating what expected condition is not satisfied